### PR TITLE
MWPW-157460 Allow Ungated Marketo One Page Experience

### PR DESCRIPTION
**Pull request by meganthecoder**
* Allows "Ungated" One Page Experience for Marketo
* [Cleanup] Updates Marketo lana logs to correct format

Add query parameter `?form=off` to any Marketo One Page Experience url to bypass the form. The form won't show up, and instead, the content (video, pdf, etc) will be immediately visible.

Resolves: [MWPW-157460](https://jira.corp.adobe.com/browse/MWPW-157460)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-html?form=off&martech=off
- After: https://methomas-one-page-ungated--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-html?form=off&martech=off

### CODE REVIEW

**Comment by meganthecoder** [comment]
Here are the other examples:
Video
- Before: https://main--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-video?form=off&martech=off
- After: https://methomas-one-page-ungated--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-video?form=off&martech=off
PDF
- Before: https://main--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-pdf?form=off&martech=off
- After: https://methomas-one-page-ungated--milo--adobecom.hlx.live/drafts/bmarshal/one-page/one-page-pdf?form=off&martech=off
Marquee
- Before: https://main--milo--adobecom.hlx.page/drafts/methomas/one-page-marquee?form=off&martech=off
- After: https://methomas-one-page-ungated--milo--adobecom.hlx.page/drafts/methomas/one-page-marquee?form=off&martech=off

A note on performance: There are known issue with performance, see ticket [MWPW-157460](https://jira.corp.adobe.com/browse/MWPW-157460). The team has decided to release this now for a small-scale test on bacom in order to gather data, and we'll revisit the issue of performance after crux data is available.

File: libs/blocks/marketo/marketo.js [hunk 1]
```diff
@@ -91,6 +92,35 @@ export const setPreferences = (formData) => {
...
+    // For Marquee use case
+    let count = 0;
+    const interval = setInterval(() => {
+      const el = document.querySelector(`.section.${section}`);
+      if (el) {
+        clearInterval(interval);
+        show(el);
+      }
+      count += 1;
+      if (count > 6) clearInterval(interval);
```
Comment (JasonHowellSlavin) [inline libs/blocks/marketo/marketo.js:116]: nit: might be more readable to set 6 to a var, like `maxIntervals` or something along those lines.

File: libs/blocks/marketo/marketo.js [hunk 2]
```diff
@@ -91,6 +92,35 @@ export const setPreferences = (formData) => {
...
+  try {
+    const show = (el) => {
+      el.classList.remove('hide-block');
+      if (scroll) el.scrollIntoView({ behavior: 'smooth' });
+    };
+    const section = formData[SUCCESS_SECTION].toLowerCase().replaceAll(' ', '-');
+    const success = document.querySelector(`.section.${section}`);
+    if (success) {
+      show(success);
+      return;
```
Comment (JasonHowellSlavin) [inline libs/blocks/marketo/marketo.js:105]: Super nit: maybe rename section to successClass and success to section, might read a bit easier: 
```suggestion
    const successClass = formData[SUCCESS_SECTION].toLowerCase().replaceAll(' ', '-');
    
    const section = document.querySelector(`.section.${successClass}`);
    if (section) {
      show(section);
      return;
```

File: test/blocks/marketo/marketo.test.js [hunk 3]
```diff
@@ -104,3 +105,41 @@ describe('marketo decorateURL', () => {
...
+    const clock = sinon.useFakeTimers();
+
+    init(document.querySelector('.marketo'));
+    expect(document.querySelector('#success-section').classList.contains('hide-block')).to.be.true;
+    document.querySelector('#success-section').classList.add('form-success');
+    clock.tick(500);
+    expect(document.querySelector('#success-section').classList.contains('hide-block')).to.be.false;
+
+    url.searchParams.delete(FORM_PARAM);
+    window.history.pushState({}, '', url);
```
Comment (JasonHowellSlavin) [inline test/blocks/marketo/marketo.test.js:142]: candidates for `afterEach`?

**Review by JasonHowellSlavin — APPROVED** [review]
Looks great! Some small nits. 

Might be good to have a test that shows the triggered `catch` statement from your showSuccessSection function that occurs on line 118 marketo/marketo.js

**Comment by meganthecoder** [comment]
> Might be good to have a test that shows the triggered `catch` statement from your showSuccessSection function that occurs on line 118 marketo/marketo.js

@JasonHowellSlavin Good point. I decided to remove the try/catch in favor of catching the errors in if statements - less nested and made more sense with the setInterval code. Added tests for those two use cases.

File: libs/blocks/marketo/marketo.js [hunk 4]
```diff
@@ -198,6 +221,15 @@ export default function init(el) {
     return;
   }
 
+  const searchParams = new URLSearchParams(window.location.search);
+  const ungated = searchParams.get(FORM_PARAM) === 'off';
+
+  if (ungated) {
```
Comment (Brandon32) [inline libs/blocks/marketo/marketo.js:227]: We should check if we are on a one page
```suggestion
  if (formData.[SUCCESS_TYPE] === 'section' && ungated) {
```

**Review by Brandon32 — APPROVED** [review]

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**